packages/@aws-cdk/aws-ecr/lib/repository.ts

```diff
@@ -319,6 +343,43 @@ export abstract class RepositoryBase extends Resource implements IRepository {
       'ecr:UploadLayerPart',
       'ecr:CompleteLayerUpload');
   }
+
+  /**
+   * Returns the account ID of the account to put into the Resource Policy
+   * if we cannot put a reference to the principal itself there,
+   * and 'undefined' in case we can.
+   */
+  private principalCannotBeSafelyAddedToResourcePolicy(grantee: iam.IGrantable): string | undefined {
```

Comment:
Can't this return the correct `IGrantable` instead?

```ts
private correctGrantable(grantee: IGrantable): IGrantable {
  if (needsToBeDifferent) {
    return new AccountRootPrincipal(correctAccount);
  }

  return grantee;
}
```

Reply:
Unfortunately, the code you suggested:

```ts
    return new AccountRootPrincipal(Stack.of(this.account), {
      conditions: {
        StringEquals: { 'aws:PrincipalTag/aws-cdk:id', this.roleTag },
      },
    });
```

doesn't work (conditions are attached to Policies, not to Principals).

However, added the tag to the Role in a different way, let me know if that works!